proposals/channel-jamming-mitigation.md

```diff
@@ -0,0 +1,257 @@
+costs of running a node, funding channels on-chain and the opportunity cost
+of capital committed to the attack.
+
+### Threat Model
+Attackers may aim to disrupt a specific set of channels, or the network as a
+whole. We consider the following threat model, adapted from [1]: 
+* The attacker can quickly set up a set of seemingly unrelated nodes and open
+  channels to any public node in the network.
+* The attacker has an up to date view of the network's public topology. 
+* The attacker is economically rational, meaning that they will pursue an 
+  attack with the lowest cost and highest utility (and "seeing the world burn" 
+  classifies as utility).
+* The attacker has the ability to send slow or quick jams: 
+  * Quick jamming attacks will aim to fill all available htlc slots if
+    `max_accepted_htlcs` * `htlc_minimum_msat` < `max_htlc_value_in_flight_msat`,
+    or otherwise will aim to deplete channel liqudidity.
+  * The attacker has a modified LN node implementation that allows them to hold
+    HLTCs up until the point of force closure, then release them.
+* The attacker may have access to long-lived channels when the attack begins.
+
+### Mitigation
+To comprehensively mitigate jamming attacks against the Lightning Network, a
+solution needs to address both quick and slow jamming. This proposal has two 
+parts, upfront fees to address quick jamming and reputation to address slow
+jamming [2].
+
+### Quick Jamming Mitigation
+Traffic that is part of a quick jamming attack mimics the behavior of honest 
+nodes' failed payments, which makes it difficult to identify. A solution that 
+addresses quick jamming therefore *must have a trivial impact on honest 
+traffic, but be sufficient to compensate the node(s) under attack*. A solution 
+that fails to do this will only magnify an attacker's ability to disrupt the 
+network, as they can leverage the jamming mitigation itself to further degrade 
+honest traffic.
+
+An upfront fee, paid regardless of the outcome of a payment attempt, is proposed
+to economically compensate nodes for providing traffic with access to liquidity
+and slots. Simulations in [2] show that a fee of as little as 1% of the success
+case fees are sufficient to compensate the opportunity cost of forwarding nodes
+in various routing scenarios.
+
+Nodes that advertise `option_upfront_fee` (56/57) will advertise a new TLV in
+their `channel_update` which expresses unconditional fees as a percentage of 
+their success-case fees advertised in the update: 
+```
+1. `tlv_stream`: `channel_update_tlvs`
+2. types:
+    1. type: 1 (`upfront_fee_policy`)
+    2. data:
+        * [`tu32`:`upfront_fee_base_ppm`]
+        * [`tu32`:`upfront_fee_proportional_ppm`]
+
+```
+
+To save the network the bandwidth and storage required to transmit and save
+defaults, nodes that advertise `option_upfront_fee` in their node announcement
+should be assumed to have a base and proportional fee that is 1% of its success
```

Comment:
It's a magic value, but if it is just a hard-coded default to potentially save bytes, it hardly hurts. If it turns out that nobody wants 1% and every channel update contains explicit upfront fees, then the only cost is the code to support the not-used default.